src/autoscaler/api/publicapiserver/public_api_handler.go

```diff
@@ -386,3 +391,66 @@ func (h *PublicApiHandler) GetApiInfo(w http.ResponseWriter, r *http.Request, va
 func (h *PublicApiHandler) GetHealth(w http.ResponseWriter, r *http.Request, vars map[string]string) {
 	w.Write([]byte(`{"alive":"true"}`))
 }
+
+func (h *PublicApiHandler) CreateCustomMetricsCredential(w http.ResponseWriter, r *http.Request, vars map[string]string) {
+	appId := vars["appId"]
+	if appId == "" {
+		h.logger.Error("AppId is missing", nil, nil)
+		handlers.WriteJSONResponse(w, http.StatusBadRequest, models.ErrorResponse{
+			Code:    "Bad Request",
+			Message: "AppId is required",
+		})
+		return
+	}
+
+	h.logger.Info("Create custom metric credential", lager.Data{"appId": appId})
+	cred, err := custom_metrics_cred_helper.CreateCustomMetricsCredential(appId, h.policydb, custom_metrics_cred_helper.MaxRetry)
```

Comment:
As discussed,  end-user can specify username/password if they like with this API.  So, request body parsing is necessary here.